File: mvc-arch/src/main/java/org/example/mvc/view/RedirectView.java
package org.example.mvc.view;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.util.Map;

public class RedirectView implements View {
    public static final String DEFAULT_REDIRECT_PREFIX = "redirect:";
    private final String redirectName;

    public RedirectView(String redirectName) {
        this.redirectName = redirectName;
    }

    @Override
    public void render(Map<String, ?> model, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String path = redirectName.substring(DEFAULT_REDIRECT_PREFIX.length());
        response.sendRedirect(path);
    }
}

File: refactor-to-kotlin/src/main/kotlin/org/example/mvc/view/RedirectView.kt
package org.example.mvc.view

import org.example.mvc.view.resolver.DEFAULT_REDIRECT_PREFIX
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

class RedirectView(private val viewName: String) : View {
    override fun render(model: Map<String, *>, req: HttpServletRequest, resp: HttpServletResponse) {
        val path = viewName.substring(DEFAULT_REDIRECT_PREFIX.length)
        resp.sendRedirect(path)
    }
}